components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts

```diff
@@ -0,0 +1,549 @@
+import { map } from 'lodash/fp';
+
+import { EntityStatus } from 'app/entities/entities';
+import { ProjectConstants } from 'app/entities/projects/project.model';
+import {
+  ProjectsFilterState,
+  projectsFilterInitialState,
+  projectsFilterReducer,
+  ProjectsFilterOption
+} from './projects-filter.reducer';
+import { LoadOptions, LoadOptionsSuccess } from './projects-filter.actions';
+
+const UNASSIGNED_PROJECT_ID = ProjectConstants.UNASSIGNED_PROJECT_ID;
+const UNASSIGNED_PROJECT_LABEL = ProjectConstants.UNASSIGNED_PROJECT_LABEL;
+const ALL_RESOURCES_LABEL = ProjectConstants.ALL_RESOURCES_LABEL;
+const ALL_PROJECTS_LABEL = ProjectConstants.ALL_PROJECTS_LABEL;
+const MULTIPLE_PROJECTS_LABEL = ProjectConstants.MULTIPLE_PROJECTS_LABEL;
+const BADGE_GREY_STATE = false;
+const BADGE_BLUE_STATE = true;
+
+describe('projectsFilterReducer', () => {
+  const initialState: ProjectsFilterState = projectsFilterInitialState;
+
+  describe('LOAD_OPTIONS', () => {
+    const action = new LoadOptions();
+
+    it('sets status to loading', () => {
+      const { optionsLoadingStatus } = projectsFilterReducer(initialState, action);
+      expect(optionsLoadingStatus).toEqual(EntityStatus.loading);
+    });
+  });
+
+  describe('LOAD_OPTIONS_SUCCESS', () => {
```

Comment:
This closely aligns to the requirements in the JIRA card. There are actually more cases here, though, filling in a few gaps in those requirements (derived from the video attached to the JIRA card).